WalletConnect/web-examples
# Feat/region relayer dropdown

**Overview**
- Worked on adding a DropDown menu for regions and selecting different relayers
- As per #57 
- [Loom Recording](https://www.loom.com/share/f22c2ccd23ff4dcc92238a61f34b6e03)

**_Note_**
As the new relayers go live (~ Wednesday), they should function to resolve the new createClient() result when live. Hence the Draft PR.

**Process**
- Understanding `ClientContext.tsx `&` Index.tsx`
- Added `regionRelayer` and `setRegionRelayer` state into the UseMemo + Interface
- Add a condition if the Relayer changes from the `DEFAULT_RELAY_URL`, a new `createClient()` will be instantiated
- Work on a new Component called `Dropdown.tsx`
- Used React-Select to make it easier to not build out from scratch. If it is preferred to not use libraries and more vanilla, I can re-create it.

**How to Reproduce**
- Use this branch + do a clean `npm i`
- Add the standard Project id details as outlined in the ReadME file
- `Yarn dev`
- Check the console logs for the correct RELAYERS have been instantiated.
